secure.go

```diff
@@ -290,6 +295,10 @@ func (s *Secure) processRequest(w http.ResponseWriter, r *http.Request) (http.He
 	}
 
 	// Allowed hosts check.
+	if s.opt.AllowedHostsFunc != nil {
+		s.opt.AllowedHosts = s.opt.AllowedHostsFunc()
```

Comment:
Can we do a second check instead of overwriting the `s.opt.AllowedHosts` value? Ideally I think we should support both the static list of allowed hosts and the new function.

Reply:
Good point, totally agree.